airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeS3StagingDestination.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.snowflake;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.sentry.AirbyteSentry;
+import io.airbyte.integrations.destination.NamingConventionTransformer;
+import io.airbyte.integrations.destination.jdbc.AbstractJdbcDestination;
+import io.airbyte.integrations.destination.s3.S3DestinationConfig;
+import io.airbyte.integrations.destination.staging.StagingConsumerFactory;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SnowflakeS3StagingDestination extends AbstractJdbcDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeS3StagingDestination.class);
+
+  public SnowflakeS3StagingDestination() {
+    this(new SnowflakeSQLNameTransformer());
+  }
+
+  public SnowflakeS3StagingDestination(final SnowflakeSQLNameTransformer nameTransformer) {
+    super("", nameTransformer, new SnowflakeSqlOperations());
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) {
```

Comment:
is there any way to DRY this with other snowflake implementations?

Reply:
yes probably good thing to tackle in another PR as you commented https://github.com/airbytehq/airbyte/pull/10866#discussion_r827575581